### PR TITLE
Fix possible bsod within log_destination_find

### PR DESCRIPTION
**Pull request by dragomirecky**

### CODE REVIEW

File: src/logging/log.c [hunk 1]
```diff
@@ -68,11 +68,12 @@ log_component_t *log_component_find(const char *name) {
 }
 
 log_destination_t *log_destination_find(const char *name) {
-    for (log_destination_t **destination_pp = &destinations_head; *destination_pp != NULL; destination_pp++) {
+    for (log_destination_t **destination_pp = &destinations_head; *destination_pp != NULL;) {
         log_destination_t *destination = *destination_pp;
         if (strcmp(destination->name, name) == 0) {
             return destination;
         }
+        destination_pp = &((*destination_pp)->next);
```
Comment (mkbel) [inline src/logging/log.c:76]: Is this equivalent to? 
log.c:49
```
for (log_destination_t **destination_pp = &destinations_head; *destination_pp != NULL; destination_pp = &(*destination_pp)->next) {
```

Comment (mkbel) [reply src/logging/log.c:76]: The same construct which is fixed here is also used there without being fixed, is this right?
```
void log_destination_unregister(log_destination_t *destination) {
    for (log_destination_t **destination_pp = &destinations_head; *destination_pp != NULL; destination_pp++)

log_component_t *log_component_find(const char *name) {
    for (log_component_t *component = __start_log_components; component < __end_log_components; component++) {
```

**Review by mkbel — COMMENTED** [review]
.

File: src/logging/log.c (hunk 1 above)
Comment (dragomirecky) [reply src/logging/log.c]: nice catch 👍 fixed the other occurrence & unified it

**Review by mkbel — DISMISSED** [review]

**Review by mkbel — APPROVED** [review]

**Pull request merged**